### ISSUE TITLE
Rename some keys in config

### ISSUE THREAD

**Issue by mathpluscode** [ucl-candi/DeepReg#86]
# Issue description

Some tests to add
- different loss scales
- different extract levels

Some changes as follows
- tf to training
- similarity to dissimilarity
- weight 0. to 0.0
- reduce test loss scales
- add comments for optimizer name

```yaml
training:
  model:
    method: "ddf" # ddf or conditional or seg
    backbone:
      name: "global"
      out_kernel_initializer: "zeros" # zeros or glorot_uniform
      out_activation: ""
    local:
      num_channel_initial: 1
      extract_levels: [0, 1, 2, 3, 4]
      pooling: true
      concat_skip: false
  loss:
    dissimilarity:
      image:
        name: "lncc"
        weight: 0.0
      label:
        weight: 1.0
        name: "multi_scale"
        multi_scale:
          loss_type: "dice"
          loss_scales: [0, 1, 2, 4, 8, 16]
        single_scale:
          loss_type: "cross-entropy"
    regularization:
    batch_size: 2
    shuffle_buffer_num_batch: 0
  opt:
    name: "adam"
    name: "adam" # "adam" / "sgd" / "rms"
    adam:
      learning_rate: 1.0e-5
```

## Type of Issue

- [ ] Bug report
- [ ] Documentation update
- [ ] Test request
